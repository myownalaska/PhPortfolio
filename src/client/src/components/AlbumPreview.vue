<template>
  <article>
    <a :href="albumLink">
      <img class="preview-image" alt="Album Cover" src="../assets/katya.jpeg"/>
    </a>
    <p class="date">{{ formattedDate }}</p>
    <h3 class="title">{{ title }}</h3>
  </article>
</template>

<script>
import { formatDate } from "../utils/formatDate";

export default {
  name: "AlbumPreview",
  props: ["album"],
  computed: {
    albumLink() {
      return `/albums/${this.album.id}`;
    },
    title() {
      return `${this.album.name}, ${this.album.location}`.toUpperCase();
    },
    formattedDate() {
      return formatDate(this.album.date);
    }
  }
};
</script>

<style scoped>
.preview-image {
  max-width: 100%;
  margin-bottom: 32px;
}

.date {
  display: block;
  margin-bottom: 8px;
  line-height: 1em;
  font-size: 18px;
  font-family: "Mukta Mahee", sans-serif;
}

.title {
  margin: 0;
  color: #b2b1b1;
  line-height: 1em;
  font-size: 18px;
  font-family: "Mukta Mahee", sans-serif;
}
</style>
