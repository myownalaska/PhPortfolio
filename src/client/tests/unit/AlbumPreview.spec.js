import { mount } from '@vue/test-utils';
import AlbumPreview from '@/components/AlbumPreview.vue';

const FAKE_ALBUM = {
  id: 1,
  name: 'Fake Album',
  description: 'It is not real',
  date: '2017-06-22',
  location: 'ST. PETERSBURG',
  model: {
    firstName: 'Your',
    lastName: 'Name',
  },
};

describe('AlbumPreview.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(AlbumPreview, { propsData: { album: FAKE_ALBUM } });
    expect(wrapper.text()).toMatchSnapshot();
  });
});
