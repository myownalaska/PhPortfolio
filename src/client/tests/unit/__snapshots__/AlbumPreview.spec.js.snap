// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumPreview.vue renders correctly 1`] = `"22 June FAKE ALBUM, ST. PETERSBURG"`;
